integration/integration.go

```diff
@@ -229,26 +165,84 @@ func (i *Integration) MarshalJSON() (output []byte, err error) {
 	return i.logger
 }
 
-// toJSON serializes integration as JSON. If the pretty attribute is
-// set to true, the JSON will be indented for easy reading.
-func (i *Integration) toJSON(pretty bool) (output []byte, err error) {
-	output, err = i.MarshalJSON()
-	if !pretty {
-		return
+// FindEntity finds ad return an entity by name. returns false if entity does not exist in the integration
+func (i *Integration) FindEntity(name string) (*Entity, bool) {
+	if i.Entities == nil {
+		return &Entity{}, false
 	}
+	for _, e := range i.Entities {
+		// nil metadata usually means the "host" entity, so keep searching
+		if e == nil || e.Metadata == nil {
+			continue
+		}
 
-	var buf bytes.Buffer
-	err = json.Indent(&buf, output, "", "\t")
-	if err != nil {
-		return nil, err
+		if e.Metadata.Name == name {
+			return e, true
+		}
 	}
+	return &Entity{}, false
+}
 
-	return buf.Bytes(), nil
+// Gauge creates a metric of type gauge
+func Gauge(timestamp time.Time, metricName string, value float64) (metric.Metric, error) {
+	return metric.NewGauge(timestamp, metricName, value)
+}
+
+// Count creates a metric of type count
+func Count(timestamp time.Time, metricName string, value float64) (metric.Metric, error) {
+	return metric.NewCount(timestamp, metricName, value)
+}
+
+// Summary creates a metric of type summary
+func Summary(timestamp time.Time, metricName string, count float64,
+	average float64, sum float64, min float64, max float64) (metric.Metric, error) {
+	return metric.NewSummary(timestamp, metricName, count, average, sum, min, max)
+}
+
+// CumulativeCount creates metric of type cumulative count
+func CumulativeCount(timestamp time.Time, metricName string, value float64) (metric.Metric, error) {
+	return metric.NewCumulativeCount(timestamp, metricName, value)
+}
+
+// Rate creates a metric of type rate
+func Rate(timestamp time.Time, metricName string, value float64) (metric.Metric, error) {
+	return metric.NewRate(timestamp, metricName, value)
+}
+
+// CumulativeRate creates a metric of type cumulative rate
+func CumulativeRate(timestamp time.Time, metricName string, value float64) (metric.Metric, error) {
+	return metric.NewCumulativeRate(timestamp, metricName, value)
+}
+
+// PrometheusHistogram creates a metric of type prometheus histogram
+func PrometheusHistogram(timestamp time.Time, metricName string, sampleCount uint64, sampleSum float64) (*metric.PrometheusHistogram, error) {
+	return metric.NewPrometheusHistogram(timestamp, metricName, sampleCount, sampleSum)
+}
+
+// PrometheusSummary creates a metric of type prometheus summary
+func PrometheusSummary(timestamp time.Time, metricName string, sampleCount uint64, sampleSum float64) (*metric.PrometheusSummary, error) {
+	return metric.NewPrometheusSummary(timestamp, metricName, sampleCount, sampleSum)
+}
+
+// -- private
+// is entity empty?
+func notEmpty(entity *Entity) bool {
+	return len(entity.Events) > 0 || len(entity.Metrics) > 0 || len(entity.Inventory.Items()) > 0
```

Comment:
we might want a `Len()` method for Inventory instead

Reply:
fixed